### ISSUE TITLE
unsupported operand type(s) for +: 'int' and 'NoneType'

### ISSUE THREAD

**Issue by tbm** [ghislainbourgeois/beancount_portfolio_allocation#1]
I'm getting:

```
bean-portfolio-allocation-report 2018.beancount --portfolio pension
Traceback (most recent call last):
  File "/home/tbm/.local/bin/bean-portfolio-allocation-report", line 11, in <module>
    sys.exit(main())
  File "/home/tbm/.local/lib/python3.5/site-packages/beancount_portfolio_allocation/allocation_report.py", line 58, in main
    print(report(args.bean, args.portfolio))
  File "/home/tbm/.local/lib/python3.5/site-packages/beancount_portfolio_allocation/allocation_report.py", line 10, in report
    targets, allocations, total = loader.load(bean, portfolio)
  File "/home/tbm/.local/lib/python3.5/site-packages/beancount_portfolio_allocation/loader.py", line 18, in load
    total = allocations.total_invested_for_portfolio()
  File "/home/tbm/.local/lib/python3.5/site-packages/beancount_portfolio_allocation/allocation.py", line 38, in total_invested_for_portfolio
    return sum([p.value for p in self])
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```

**Comment by tbm** [comment]
Here's a test case:

```
option "operating_currency" "USD"
option "inferred_tolerance_default" "USD:0.01"

1970-01-01 open Equity:Opening-Balance

1970-01-01 open Assets:Pension
  portfolio: "pension"

1970-01-01 open Assets:Pension:NEST
  portfolio: "pension"

1970-01-01 commodity AAPL
  asset-class: "equity"
  asset-subclass: "usa"

1970-01-01 commodity NESTHIGHER
  asset-class: "equity"
  asset-subclass: "uk"

1970-01-01 commodity USD
  asset-class: "cash"
  asset-subclass: "cash"

2017-12-29 price NESTHIGHER 2.2016 GBP
2018-03-29 price NESTHIGHER 2.0856 GBP

2018-01-01 * "Opening balance"
  Assets:Pension            100 AAPL {1000 USD}
  Equity:Opening-Balance

2018-01-01 * "Opening balance: NEST Higher Risk Fund" #opening-balance
  Assets:Pension                   11.7230 NESTHIGHER {{25.61 GBP, 2017-12-14}}
  Equity:Opening-Balance
```

**Comment by ghislainbourgeois** [comment]
Thanks for the report and test case. I have played a bit with it, and it seems related to the multiple currencies and missing prices.

**Comment by ghislainbourgeois** [comment]
Latest commit 438fd276ac53711813a4a175deb1035d4ff49dad contains a partial fix for this test case. It still does not return any useful report for 2 reasons:

1. There is no price available for AAPL, but now an error message is output instead of a crash. This can be fixed by adding a price directive or using the implicit_prices plugin.

1. Multiple currency support is not yet implemented. I know how to implement it, but it will still need a conversion rate present in the directives.

**Comment by ghislainbourgeois** [comment]
Just released version 0.2.0 with support for multiple currencies. It will convert all values to the first operating_currency in your input file if you have all the required price entries. If price entries are missing or operating_currency, it will output an error message.

I plan on adding an optional parameter to select the currency to convert to in the near future. In the meantime, please let me know if this works better for you.

**Comment by tbm** [comment]
Looks better, thank you. I'll play around with it some more tomorrow.
